Make release of v1.14.2 failed

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:
Make release of v1.14.2 failed. After facing the error `hack/lib/golang.sh: line 295: KUBE_SERVER_PLATFORMS[@]: unbound variable` ([link to code](https://github.com/kubernetes/kubernetes/blob/v1.14.2/hack/lib/golang.sh#L295)) we did 4 `cherry-pick`:

- 38b7f2125fa571bdd9fe3070b808635f88586dd1
- 7e2aa46f64a67a50e8155538750c08b83dffa348
- eaafa5952f8073f51b13dbeafe406019995c8c0b

- a commit in @cblecker 's opening PR #78048 (b8481b62fcb38d7af96d4af2e2077fcfb2921712)

Then the build process failed again:
```
/home/ubuntu/kubernetes/hack/lib/golang.sh: line 305: KUBE_SERVER_PLATFORMS[@]: unbound variable
/home/ubuntu/kubernetes/hack/lib/golang.sh: line 305: KUBE_SERVER_PLATFORMS[@]: unbound variable
Makefile.generated_files:514: recipe for target 'gen_bindata' failed
make[2]: *** [gen_bindata] Error 1
make[2]: Leaving directory '/home/ubuntu/kubernetes'
Makefile:526: recipe for target 'generated_files' failed
make[1]: *** [generated_files] Error 2
make[1]: Leaving directory '/home/ubuntu/kubernetes'
Makefile:23: recipe for target 'kubernetes-build' failed
make: *** [kubernetes-build] Error 2
```
What we did was:
we ran two commands
```
make generated_files
make WHAT=cmd/kubectl KUBE_BUILD_PLATFORMS="darwin/amd64 windows/amd64"
```
`make generated_files` called [kubernetes/build/root/Makefile](https://github.com/kubernetes/kubernetes/blob/v1.14.2/build/root/Makefile#L92), after that,  we did `make WHAT=cmd/kubectl KUBE_BUILD_PLATFORMS="darwin/amd64 windows/amd64"`. And right after this command, the error happened.

**Environment**:
- Kubernetes version (use `kubectl version`): v1.14.2
- OS (e.g: `cat /etc/os-release`): ubuntu 16.04, bash version 4.3

@cblecker @tpepper @figo

--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -302,7 +302,7 @@ kube::golang::server_test_targets() {
 IFS=" " read -ra KUBE_TEST_SERVER_TARGETS <<< "$(kube::golang::server_test_targets)"
 readonly KUBE_TEST_SERVER_TARGETS
 readonly KUBE_TEST_SERVER_BINARIES=("${KUBE_TEST_SERVER_TARGETS[@]##*/}")
-readonly KUBE_TEST_SERVER_PLATFORMS=("${KUBE_SERVER_PLATFORMS[@]}")
+readonly KUBE_TEST_SERVER_PLATFORMS=("${KUBE_SERVER_PLATFORMS[@]:+"${KUBE_SERVER_PLATFORMS[@]}"}")
 
 # Gigabytes necessary for parallel platform builds.
 # As of January 2018, RAM usage is exceeding 30G

